test/test-rotateElement.js

```diff
@@ -0,0 +1,414 @@
+            -----
+            ----X
+            -----
+            -----
+            -----
+        `)
+        assert.deepEqual(expectedResult[0], stepElementTetromino)
+    })
+})
+
+describe("Rotation where starts to the bottom, pivot has different x and y coordinates", function () {
+    it("element move the left field 6x6", function () {
+        let elementTetromino = tetris.getCoordinates(`
+            ------
+            ------
+            ------
+            ------
+            ------
+            ---X--
+        `)
+        let pivot = tetris.getCoordinates(`
+            ------
+            ------
+            --X---
+            ------
+            ------
+            ------
+        `)
+        let stepElementTetromino = tetris.rotateElement(elementTetromino[0], pivot[0])
+        let expectedResult = tetris.getCoordinates(`
+            ------
+            -----X
+            ------
+            ------
+            ------
+            ------
+        `)
+        assert.deepEqual(expectedResult[0], stepElementTetromino)
+    })
+
+    it("element move the top field 6x6", function () {
+        let elementTetromino = tetris.getCoordinates(`
+            ------
+            -----X
+            ------
+            ------
+            ------
+            ------
+        `)
+        let pivot = tetris.getCoordinates(`
+            ------
+            ------
+            --X---
+            ------
+            ------
+            ------
+        `)
+        let stepElementTetromino = tetris.rotateElement(elementTetromino[0], pivot[0])
+        let expectedResult = { x: 1, y: -1 }
+        assert.deepEqual(expectedResult, stepElementTetromino)
+    })
+
+    it("element move the top field 6x6", function () {
+        let elementTetromino = { x: 1, y: -1 }
+        let pivot = tetris.getCoordinates(`
+            ------
+            ------
+            --X---
+            ------
+            ------
+            ------
+        `)
+        let stepElementTetromino = tetris.rotateElement(elementTetromino, pivot[0])
+        let expectedResult = { x: -1, y: 3 }
+        assert.deepEqual(expectedResult, stepElementTetromino)
+    })
+    it("element move the top field 6x6", function () {
+        let elementTetromino = { x: -1, y: 3 }
+        let pivot = tetris.getCoordinates(`
+            ------
+            ------
+            --X---
+            ------
+            ------
+            ------
+        `)
+        let stepElementTetromino = tetris.rotateElement(elementTetromino, pivot[0])
+        let expectedResult = tetris.getCoordinates(`
+            ------
+            ------
+            ------
+            ------
+            ------
+            ---X--
+        `)
+        assert.deepEqual(expectedResult, stepElementTetromino)
+    })
+})
+
+describe("Rotation where pivot coordinates coincides with element coordinates", function () {
+    it("element not move field 4x4", function () {
```

Comment:
I do not understand
pivot === elementTetromino
and
expectedResult === pivot
too
What is wrong here?

Reply:
Pleaese google "indent" and make sure all 3 of these variables have the same indent